Sources/BuildSystemIntegration/ExternalBuildSystemAdapter.swift

```diff
@@ -226,19 +225,17 @@ actor ExternalBuildSystemAdapter {
         try? FileManager.default.contentsOfDirectory(at: buildServerConfigLocation, includingPropertiesForKeys: nil)
         .filter { $0.pathExtension == "json" }
 
-      if let configFileURL = jsonFiles?.sorted(by: { $0.lastPathComponent < $1.lastPathComponent }).first,
-        let configFilePath = AbsolutePath(validatingOrNil: try? configFileURL.filePath)
-      {
-        return configFilePath
+      if let configFileURL = jsonFiles?.sorted(by: { $0.lastPathComponent < $1.lastPathComponent }).first {
+        return configFileURL
       }
     }
 
     // Pre Swift 6.1 SourceKit-LSP looked for `buildServer.json` in the project root. Maintain this search location for
     // compatibility even though it's not a standard BSP search location.
     if let workspaceFolder = workspaceFolder,
-      localFileSystem.isFile(workspaceFolder.appending(component: "buildServer.json"))
+      FileManager.default.fileExists(at: workspaceFolder.appendingPathComponent("buildServer.json"))
```

Comment:
There is a `FileManager.fileExists(atPath:isDirectory:)` which can help ensure that the path exists and is a file (`!bIsDirectory`).

Reply:
Good suggestion, I even wrote a convenience method for that in this PR `FileManager.isFile`. Though I think it’s unlikely that somebody would have a directory named `buildServer.json` 😉